### PR TITLE
Avoid \pnum inside examples.

### LINKED ISSUE #781: \pnums in examples and notes are ugly and confusing
Update: the issue with \pnums in effects/return/etc. clauses was separated out and is now issue #821.

Adding \pnums added in the middle of examples and notes makes it hard to see where the example/note starts/ends, and \pnums added in \effects (or other) clauses causes the wording to spill over into a new paragraph making it look like it's no longer part of the Effects (or other).

As an example of \pnums in an example, have a look at 27.10.8.6.2 [path.factory]/p4 - it has new paragraphs as they appeared in the original wording, but it is ugly!  See also the Note above in 27.10.8.6 [path.non-member]/p8.

Inserting a blank line in the Latex before the text in the examples/notes/effects/etc. seems to work - is there a better way?  In any case, I think we should fix these as we come across them.  Or in one big swoop if someone has the patience.

Having examples embedded in the paragraph is ugly also - see 1.9 [intro.execution]/p16.  It would be nice to address those while we're at it, perhaps also by inserting a blank line before/after the example?  \linebreak doesn't work because it causes the previous line to get stretched, and \linebreak after a blank line results in an error.

Here's a few examples of embedded \pnums that I think we may want to fix:
- \pnums in clauses:
  *\* [ios.members.static]/p3
  *\* [streambuf.virt.pback]/p3
  *\* [istream::sentry]/p3
  *\* [istream::extractors]/p7
  *\* [istream::extractors]/p14
  *\* [istream.unformatted]/p8
  *\* [istream.unformatted]/p14
  *\* [istream.unformatted]/p18
  *\* [ostream.inserters.arithmetic]/p1
  *\* [ostream.inserters]/p7 (this is as far as I got thru Clause 27)
- \pnums in Examples:
  *\* [path.factory]/p4
  *\* [fs.op.copy]/p6
  *\* [fs.op.system_complete]/p5
  *\* [fs.op.temp_dir_path]/p3
- \pnums in Notes:
  *\* [path.non-member]/p8

Here's a few examples of embedded exampes/notes that I think we may want to fix:
- examples embedded in the paragraphs:
  *\* [intro.execution]/p16
- notes embedded in the paragraphs:
  *\*  [path.io]/p1

### PR DESCRIPTION
**Pull request by jensmaurer**
With the exception of [facets.examples].

Fixes #781.

### CODE REVIEW

File: source/access.tex [hunk 1]
```diff
@@ -116,8 +116,9 @@
 the class appearing outside the class's \grammarterm{member-specification}{.}
 \begin{note} This access also applies to implicit references to constructors,
 conversion functions, and destructors. \end{note}
-\begin{example}
 
+\pnum
+\begin{example}
```
Comment (tkoeppe) [inline source/access.tex]: Consider here (and everywhere) whether the example illustrates the note or the section in general. (I'll take a look, too, but it's surprisingly hard to make GitHub show more context of the change.)

Comment (jensmaurer) [inline source/access.tex]: There a star-like thing on the upper left corner of each diff hunk that, when pressed, shows more context.

Anyway, I did consider for each change whether I should create a new numbered paragraph for an example or not, and I decided on an individual basis. For this particular case, the example clearly does not belong to the preceding note, and I figured a separate paragraph would be more appropriate.

Comment (tkoeppe) [inline source/access.tex]: Yeah, the star doesn't work well for interior changes :-S

OK, good, if this is all analyzed case-by-case, then I'm happy with it.

File: source/declarations.tex [hunk 2]
```diff
@@ -3918,14 +3918,12 @@
 }
 \end{codeblock}
 
-\pnum
 The \tcode{carries_dependency} attribute on function \tcode{f} means that the
```
Comment (tkoeppe) [inline source/declarations.tex]: (To be clear, these *are* still paragraph breaks, they're just not numbered. I'd like to call these "underhanded paragraphs". We do this in a few places.)

Comment (jensmaurer) [inline source/declarations.tex]: Fully agreed. For this and a few other cases, I saw no way other than removing the number, but keeping the paragraph break.

**Pull request merged**